runtime/oti/j9nonbuilder.h

```diff
@@ -4885,6 +4885,7 @@ typedef struct J9VMContinuation {
 	UDATA* stackOverflowMark;
 	UDATA* stackOverflowMark2;
 	J9JavaStack* stackObject;
+	UDATA pinCount;
```

Comment:
why is this seperate from the others?

Reply:
> or perhaps, I'm misinterpreting this. Is that to track to number of times the continuation has been pinned?

Yes. Will `J9VMThread` maintain a list of `Continuations`?

Reply:
tracking continuation lineage can be done in java code (ie. continuation.parent/child). 

> I don't see linkNext and linkPrevious fields in J9VMContinuation

This was intended for the purposes of finding all continuations in core dumps

Reply:
> Because it is a property of the continuation. Others are properties of the thread.

okay, sounds like there are 3 categories. native, synchronization and continuation.pin/unpin?

I still think all 3 can go on the thread. There is no reason for an unmounted continuation to ever have a `pinCount > 0`

Reply:
Can continuations be nested? If so, a thread can be associated to multiple continuations. In such a case, J9VMThread counter won't work; we will require a counter per continuation.

Reply:
https://cr.openjdk.java.net/~rpressler/loom/Loom-Proposal.html does mention of "nested continuations" == "delimited continuations with multiple named prompts" == "scoped continuations".

Reply:
> Can continuations be nested? If so, a thread can be associated to multiple continuations. In such a case, a J9VMThread counter won't work; we will require a counter per continuation.

Yet to find a use case where that occurs in the current Preview. Although, I am aware that it has been discussed in the past.

> Will continuations be exposed as a public API?

its currently in jdk.internal... so I would assume not for this iteration

That article is from 2018 so a lot has changed since then.

Reply:
Found a test for nested continuations: https://github.com/ibmruntimes/openj9-openjdk-jdk19/blob/openj9/test/jdk/jdk/internal/vm/Continuation/Scoped.java

Reply:
The pin/unpin natives pin the current continuation (and in the case of nesting, the most deeply nested one), which will always be the one with the live state on the j9vmthread.

Reply:
Updated as suggested. Note: RI handles nested continuations in its current implementation. Our current implementation won't be able to handle the below example. We are forcing a 1-to-1 relationship between a `ContinuationScope` and `Continuation`.

```
public static boolean isPinned(ContinuationScope scope) {

ContinuationScope X { Continuation A, Continuation B }

Continuation A { //Pinned    
   ...
   Continuation B {
      ...
   }
   ...
}
```

Reply:
Since RI currently handles nested continuations, should we do the same?

Reply:
We will, but for the purposes of unblocking JIT work I think we can keep this PR to the minimal set of changes we need right now